### DISCUSSION TITLE
Multipart Upload http-client implementation

### DISCUSSION THREAD

**Discussion by omar391** [reactphp/reactphp#471] [General]
Dear Team,

I was looking for a "Streaming Multipart" upload on the client side. Then found this: https://github.com/reactphp/http/pull/200 and then this: https://github.com/reactphp/http/issues/428#issuecomment-1030585707.

However, to my surprise, I could not find a possible implementation example of a "Non-streaming" case of "multi-part upload" on the client side.

Could anybody direct me in the correct direction for this?

Note: Thanks to @SimonFrings for mentioning this discussion thread!

**Comment by SimonFrings** [comment]
Hey @omar391, thanks for opening this discussion :+1: 

If you want to send a multipart request with ReactPHP, you have to configure your headers and body manually to fit the right format. There's currently no automatic way (builder/parser) for this, but I can definitely see this as a future addition to this project. You can also take a look at https://docs.php-http.org/en/latest/components/multipart-stream-builder.html or other multipart builder, just keep in mind that the body on ReactPHP client side has to be a  `string` or `ReadableStreamInterface`.

Does this help to you get started?
